### PR TITLE
Check for invalid hash algorithm name

### PR DESCRIPTION
**Pull request by karan6181**
## Description of changes:
- Added a check for invalid hash algorithm name.
- Added the test case for the same.

## Merge Checklist:
_Put an `x` without space in the boxes that apply. If you are unsure about any checklist, please don't hesitate to ask. We are here to help! This is simply a reminder of what we are going to look for before merging your pull request._

### General
- [x] I have read the [contributor guidelines](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md)
- [ ] This is a documentation change or typo fix. If so, skip the rest of this checklist.
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the MosaicML team.
- [x] I have updated any necessary documentation, including [README](https://github.com/mosaicml/streaming/blob/main/README.md) and [API docs](https://github.com/mosaicml/streaming/tree/main/docs) (if appropriate).

### Tests
- [x] I ran `pre-commit` on my change. (check out the `pre-commit` section of [prerequisites](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#prerequisites))
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate).
- [x] I ran the tests locally to make sure it pass. (check out [testing](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#submitting-a-contribution))
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes.

### CODE REVIEW

File: streaming/base/stream.py [hunk 1]
```diff
@@ -327,6 +327,11 @@ def _decompress_shard_part(self, zip_info: FileInfo, zip_filename: str, raw_file
 
         # Validate what was downloaded.
         if self.validate_hash:
+            if self.validate_hash not in zip_info.hashes:
+                raise ValueError(
+                    f'Hash algorithm `{self.validate_hash}` provided for data ' +
```
Comment (knighton) [inline streaming/base/stream.py:332]: First "provided" -> chosen (so we have a different verb than the second provided)?
"do not" -> "does not"

File: streaming/base/stream.py [hunk 2]
```diff
@@ -392,6 +397,11 @@ def _prepare_shard_part(self,
 
                 # Validate.
                 if self.validate_hash:
+                    if self.validate_hash not in raw_info.hashes:
+                        raise ValueError(
+                            f'Hash algorithm `{self.validate_hash}` provided for data ' +
```
Comment (knighton) [inline streaming/base/stream.py:402]: Same as above

**Review by knighton — APPROVED** [review]

File: streaming/base/stream.py (hunk 1 above)
Comment (karan6181) [reply streaming/base/stream.py]: Done.

File: streaming/base/stream.py (hunk 2 above)
Comment (karan6181) [reply streaming/base/stream.py]: Done.

**Pull request merged**